### PR TITLE
[SW-1961] Get transform and transformSchema method of mojo models aligned

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

**Comment by mn-mikke** [comment]
The prediction column must be nullable because of utilization of UDFs:
```
scala> val df = spark.range(100)
scala> df.printSchema()
root
 |-- id: long (nullable = false)
val u = udf[Long, Long]{l: Long => l + 1}
scala> val resultDF = df.withColumn("result", u('id))
scala> resultDF.printSchema
root
 |-- id: long (nullable = false)
 |-- result: long (nullable = true)
```

**Comment by jakubhava** [comment]
I see, so the UDF generally changes the nullability field to `false`?

**Comment by mn-mikke** [comment]
@jakubhava It generally changes nullability, but to `true`

**Comment by jakubhava** [comment]
Ah sorry, miss-typed. Thanks!

**Comment by mn-mikke** [comment]
Note: I skipped testing `transformSchema` for `DimReduction`  since H2O's SVM currently doesn't support MOJOs

**Comment by jakubhava** [comment]
Tests looks better, but still 2 failing:

```
ai.h2o.sparkling.ml.algos.OrdinalPredictionTestSuite > transformSchema with details returns expected result FAILED

[2020-02-19T20:32:55.583Z]     org.scalatest.exceptions.TestFailedException at OrdinalPredictionTestSuite.scala:30

[2020-02-19T20:34:16.964Z] 

[2020-02-19T20:34:16.964Z] ai.h2o.sparkling.ml.models.H2OMOJOPipelineModelTestSuite > Transform and transformSchema methods are aligned - namedMojoOutputColumns is enabled FAILED

[2020-02-19T20:34:16.964Z]     org.scalatest.exceptions.TestFailedException at H2OMOJOPipelineModelTestSuite.scala:284
```

File: scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOPredictionAutoEncoder.scala [hunk 1]
```diff
@@ -27,21 +27,21 @@ trait H2OMOJOPredictionAutoEncoder {
 
   def getAutoEncoderPredictionUDF(): UserDefinedFunction = {
     if (getWithDetailedPredictionCol()) {
-      udf[Base, Row] { r: Row =>
+      udf[Detailed, Row] { r: Row =>
```
Comment (jakubhava) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOPredictionAutoEncoder.scala]: Good catch! Thanks for writing all these tests to cover all prediction outputs!

**Comment by jakubhava** [comment]
Feel free to merge when green

**Pull request merged**